cli/command/registry/login.go

```diff
@@ -47,6 +51,34 @@ func runLogin(dockerCli command.Cli, opts loginOptions) error {
 	ctx := context.Background()
 	clnt := dockerCli.Client()
 
+	if opts.password != "" {
+		fmt.Fprintf(os.Stderr, "Using --password via the CLI is insecure. Please use --password-file.\n")
+	}
+
+	if opts.passwordFile != "" {
+		contents, err := ioutil.ReadFile(opts.passwordFile)
+		if err != nil {
+			return err
+		}
+
+		/* Trim off the last \n in the file, if it exists. Most people
+		 * don't have \ns in their password, and this allows stuff like
+		 * echo "password" > foo, without having to remember to pass
+		 * -n, or vi, which can be configured to automatically append
+		 * newlines, etc.
+		 *
+		 * For users that do have a \n as the last character of their
+		 * password, they need to store it as \n\n. I think this
+		 * conforms to the principle of least surprise, but I could be
+		 * wrong :)
+		 */
+		if contents[len(contents)-1] == '\n' {
```

Comment:
Anytime I have a long comment like this I think it's a good idea to split it into a new function. That way the comment can serve as a docstring on the function instead of cluttering up the logic within a function. Even if the function is only 3 lines, that's fine.

Maybe something like `normalizePasswordFileContent()` ?